config/generic-ultimaker-ultimainboard-v2.cfg

```diff
@@ -31,18 +31,19 @@ step_pin: PC2
 dir_pin: !PC1
 enable_pin: !PC3
 microsteps: 16
-rotation_distance: 40
+rotation_distance: 16
```

Comment:
Can i ask why you have changed this to 16?

Thanks
James

Reply:
Setting it to the "correct" value of 8mm according to [this guide](https://www.klipper3d.org/Rotation_Distance.html#axes-with-a-lead-screw) leads to the Z axis moving at double speed on Ultimaker 2+. Therefore I set it to 16mm.

There's clearly something I don't understand about how to calculate that value correctly or how klipper handles these values. With `rotation_distance: 16` moving by 8mm causes one full rotation of the leadscrew.

Reply:
ok i have done some research on this, originally this file was commited with a step_distance of 0.005(these values being the precursor to rotation_distance) this was then updated later to be a rotation_distance of 8 and a microsteps of 16, which i think was wrong, as rotation_distance = <full_steps_per_rotation> * <microsteps> * <step_distance> so 200 * 16 * 0.005 actually equals 16 which works for you. but the tangle i found was that the ultimaker board actually might use 8 microsteps on Z, so in that case 200 * 8 * 0.005 = 8, so i would expect rotation_distance: 8 and microsteps: 8. I do not have this board so i cannot confirm what it comes from factory jumpered for on Z for microsteps. How this all became rotation_distance: 40 i am not sure, i cannot find the commit that did this in the history.

So... if you can confirm the microsteps set on the board, assuming its stock, we can get the right values in place :)

Thanks
James

Reply:
https://github.com/MarlinFirmware/Configurations/blob/abb24105b39f593a093baa5477310d5043682b1c/config/examples/Ultimaker/Ultimaker%202%2B/Configuration.h#L1239
https://github.com/MarlinFirmware/Configurations/blob/abb24105b39f593a093baa5477310d5043682b1c/config/examples/Ultimaker/Ultimaker%202%2B/Configuration_adv.h#L1393

Marlin uses 200/16